### ISSUE TITLE
Feature request: Slack alert threading

### ISSUE THREAD

**Issue by MaT1g3R** [prometheus/alertmanager#2165]
It would be nice to have firing alerts and resolved alerts to be grouped into a single slack thread to reduce noise and make it easier to identify currently open issues.

https://stackoverflow.com/questions/56667592/aggregate-alertmanager-messages-as-slack-threads

**Comment by simonpasquier** [comment]
The existing Slack notifier relies on Slack's [incoming webhooks](https://api.slack.com/messaging/webhooks) which support simple messages only. I'm not sure how this could work without a major overhaul of the code.

**Comment by MaT1g3R** [comment]
I think we'd have to implement the alerts as an slack app instead for this to work. If this is not a high priority for the maintainers to implement, I can take a poke at it when I have some free time.

**Comment by simonpasquier** [comment]
Yes I don't think that any of the maintainers has time to look into this. I'd encourage to prototype something using the webhook receiver so you can experiment and move faster.

**Comment by ALA-DQ** [comment]
As it is written in the section "Post your message as a reply in a thread" of the [Slack Incoming Webhooks documentation](https://api.slack.com/messaging/webhooks), it seems to be possible to send "Reply" message in Slack Thread using Incoming WebHook:

> You can use Incoming Webhooks to make your message appear as a reply in a thread. 

In addition, in the section "Replying to your message" of the [Slack message sending documentation](https://api.slack.com/messaging/sending#threading), if Alertmanager can handle the `ts` field for firing alerts and use it in the resolved, we can manage to do feature.

What do you think about this ? Am I totally wrong or is there a way to have this ?

**Comment by brian-brazil** [comment]
That'd require state from the response of a previous notification to be kept around, which isn't really possible - especially considering clustering. Each notification is independent.

**Comment by willianpaixao** [comment]
Related and possible duplicate: https://github.com/prometheus/alertmanager/issues/3221

**Comment by siavashs** [comment]
Please follow the #3221 for further discussion of this topic.
